### PR TITLE
[test][PyTorch][sagemaker] SM local unskip tests and fix mnist test

### PR DESCRIPTION
**Pull request by tejaschumbalkar**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description

- Unskip SM local test for PyTorch
- Skip mnist test for PT1.9 as there are compatibility issues with opencv. Refer https://github.com/pytorch/pytorch/issues/63441. 

## Label Checklist
- [x] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true` or `neuron_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by arjkesh** [comment]
There are all passing, is this good to go?

File: test/sagemaker_tests/pytorch/training/resources/mnist/mnist.py [hunk 1]
```diff
@@ -24,7 +24,6 @@
 import os
 import sys
 
-import cv2 as cv
```
Comment (arjkesh) [inline test/sagemaker_tests/pytorch/training/resources/mnist/mnist.py:27]: is this package the cause of the hanging tests?

File: test/sagemaker_tests/pytorch/training/resources/mnist/mnist.py [hunk 2]
```diff
@@ -242,8 +241,6 @@ def assert_can_track_sagemaker_experiments():
 
 
 if __name__ == '__main__':
-    # test opencv
```
Comment (arjkesh) [inline test/sagemaker_tests/pytorch/training/resources/mnist/mnist.py:245]: if we remove this check, do we have a different way to test opencv?

File: test/sagemaker_tests/pytorch/training/resources/mnist/mnist.py (hunk 1 above)
Comment (tejaschumbalkar) [reply test/sagemaker_tests/pytorch/training/resources/mnist/mnist.py:27]: yes

File: test/sagemaker_tests/pytorch/training/resources/mnist/mnist.py (hunk 2 above)
Comment (tejaschumbalkar) [reply test/sagemaker_tests/pytorch/training/resources/mnist/mnist.py:245]: not for PT, but we do have for TF, [test_tensorflow_opencv](https://github.com/aws/deep-learning-containers/blob/3417251ebb9ad9888ece0fe2440469cd2f44aa23/test/dlc_tests/ec2/tensorflow/training/test_tensorflow_training.py#L116)

**Review by saimidu — COMMENTED** [review]
LGTM. Can approve once temporary changes are reverted.

I have also completed the Label Checklist section on the PR Description.

**Review by saimidu — APPROVED** [review]

**Pull request closed without merging**